crates/polars-plan/src/dsl/function_expr/strings.rs

```diff
@@ -888,15 +888,25 @@ fn replace_all<'a>(
                 "replacement value length ({}) does not match string column length ({})",
                 len_val, ca.len(),
             );
-            let literal = literal || is_literal_pat(&pat);
 
-            if literal {
+            let literal_pat = literal || is_literal_pat(&pat);
+
+            if literal_pat {
                 pat = escape(&pat)
             }
 
             let reg = Regex::new(&pat)?;
 
-            let f = |s: &'a str, val: &'a str| reg.replace_all(s, val);
+            let f = |s: &'a str, val: &'a str| {
+                // According to the docs for replace_all
+                // when literal = True then capture groups are ignored.
+                if literal {
+                    reg.replace_all(s, NoExpand(val))
+                } else {
+                    reg.replace_all(s, val)
+                }
+            };
+
```

Comment:
Still a bit new at Rust. I ended up putting an `if` in the lambda when what I really wanted was to create the lambda I need once based on the `literal`. This is probably slightly slower, but I doubt it matters.

Reply:
The lambda's will be different. So you must either put them behind an indirection or call `iter_and_replace` in the branches.

But here it won't really matter indeed.